packages/lodestar/src/chain/regen/queued.ts

```diff
@@ -52,72 +53,125 @@ export class QueuedStateRegenerator implements IStateRegenerator {
+    }
   }
 
-  async getCheckpointState(cp: phase0.Checkpoint): Promise<CachedBeaconState<allForks.BeaconState>> {
-    // First attempt to fetch the state from cache before queueing
-    const checkpointState = this.checkpointStateCache.get(cp);
-    if (checkpointState) {
-      return checkpointState;
+  async getCheckpointState(
+    cp: phase0.Checkpoint,
+    rCaller: IRegenCaller
+  ): Promise<CachedBeaconState<allForks.BeaconState>> {
+    const {caller} = rCaller || {};
+    this.metrics?.regenFnCallTotal.inc({caller, entrypoint: RegenFnName.getCheckpointState});
+    const rmetrics = this.regen.getRegenFnMetrics({caller, entrypoint: RegenFnName.getCheckpointState});
+    try {
+      // First attempt to fetch the state from cache before queueing
+      const checkpointState = this.checkpointStateCache.get(cp, rmetrics);
+      if (checkpointState) {
+        return checkpointState;
+      }
+      // The state is not immediately available in the cache, enqueue the job
+      return this.jobQueue.push({key: "getCheckpointState", args: [cp, rCaller]});
+    } catch (e) {
+      this.metrics?.regenFnTotalErrors.inc({caller, entrypoint: RegenFnName.getCheckpointState});
+      throw e;
     }
-    // The state is not immediately available in the cache, enqueue the job
-    return this.jobQueue.push({key: "getCheckpointState", args: [cp]});
   }
 
-  async getBlockSlotState(blockRoot: Root, slot: Slot): Promise<CachedBeaconState<allForks.BeaconState>> {
-    return this.jobQueue.push({key: "getBlockSlotState", args: [blockRoot, slot]});
+  async getBlockSlotState(
+    blockRoot: Root,
+    slot: Slot,
+    rCaller: IRegenCaller
+  ): Promise<CachedBeaconState<allForks.BeaconState>> {
+    const {caller} = rCaller || {};
+    this.metrics?.regenFnCallTotal.inc({caller, entrypoint: RegenFnName.getBlockSlotState});
+    return this.jobQueue.push({key: "getBlockSlotState", args: [blockRoot, slot, rCaller]});
   }
 
-  async getState(stateRoot: Root): Promise<CachedBeaconState<allForks.BeaconState>> {
-    // First attempt to fetch the state from cache before queueing
-    const state = this.stateCache.get(stateRoot);
-    if (state) {
-      return state;
+  async getState(stateRoot: Root, rCaller: IRegenCaller): Promise<CachedBeaconState<allForks.BeaconState>> {
+    const {caller} = rCaller || {};
+    this.metrics?.regenFnCallTotal.inc({caller, entrypoint: RegenFnName.getState});
+    const rmetrics = this.regen.getRegenFnMetrics({caller, entrypoint: RegenFnName.getState});
+    try {
+      // First attempt to fetch the state from cache before queueing
+      const state = this.stateCache.get(stateRoot, rmetrics);
+      if (state) {
+        return state;
+      }
+      // The state is not immediately available in the cache, enqueue the job
+      return this.jobQueue.push({key: "getState", args: [stateRoot, rCaller]});
+    } catch (e) {
+      this.metrics?.regenFnTotalErrors.inc({caller, entrypoint: RegenFnName.getPreState});
```

Comment:
Isn't this metric duplicated with the code in jobQueueProcessor? Same for all the methods

Reply:
no, this counts any error thrown before it can be queued, once queued, this will be a successful return and any errors there will be caught mutually exclusively.

Reply:
I would remove anyway, the chance of getting an error here is very low